connector/spanmetricsconnector/connector.go

```diff
@@ -390,8 +398,22 @@ func (p *connectorImp) addExemplar(span ptrace.Span, duration float64, h metrics
 
 type resourceKey [16]byte
 
+func (p *connectorImp) createResourceKey(attr pcommon.Map) resourceKey {
+	if len(p.resourceMetricsKeyAttributes) == 0 {
+		h := pdatautil.MapHash(attr)
+		return resourceKey(h)
+	}
+	m := pcommon.NewMap()
+	attr.CopyTo(m)
+	m.RemoveIf(func(k string, v pcommon.Value) bool {
+		return !p.resourceMetricsKeyAttributes[k]
+	})
+	h := pdatautil.MapHash(m)
+	return resourceKey(h)
```

Comment:
nits: 
- The type cast to `resourceKey` isn't needed because they're the same types, so it can be inferred.
- We can simplify a little by just returning the result of `pdatautil.MapHash`.
- The `v` variable isn't used so to make that explicit, we can use the `_` placeholder.

```suggestion
	if len(p.resourceMetricsKeyAttributes) == 0 {
		return pdatautil.MapHash(attr)
	}
	m := pcommon.NewMap()
	attr.CopyTo(m)
	m.RemoveIf(func(k string, _ pcommon.Value) bool {
		return !p.resourceMetricsKeyAttributes[k]
	})
	return pdatautil.MapHash(m)
```